ui/components/src/StatusHistoryChart/StatusHistoryChart.tsx

```diff
@@ -0,0 +1,167 @@
+  VisualMapComponent,
+  GridComponent,
+  DatasetComponent,
+  TitleComponent,
+  TooltipComponent,
+  CanvasRenderer,
+]);
+
+export type StatusHistoryData = [number, number, number | undefined];
+
+export interface StatusHistoryChartProps {
+  height: number;
+  data: StatusHistoryData[];
+  xAxisCategories: number[];
+  yAxisCategories: string[];
+  legend?: LegendComponentOption;
+  timeScale?: TimeScale;
+}
+
+export function StatusHistoryChart(props: StatusHistoryChartProps) {
+  const { height, data, xAxisCategories, yAxisCategories, timeScale } = props;
+  const { timeZone } = useTimeZone();
+  const chartsTheme = useChartsTheme();
+  const theme = useTheme();
+
+  const uniqueValues = useMemo(
+    () => [...new Set(data.map((item) => item[2]))].filter((value): value is number => value !== undefined),
+    [data]
+  );
+
+  // get colors from theme and generate colors if not provided
+  const pieces = useMemo(() => {
+    const themeColors = Array.isArray(chartsTheme.echartsTheme.color)
+      ? chartsTheme.echartsTheme.color.filter((color): color is string => typeof color === 'string')
+      : [];
+
+    return uniqueValues.map((value, index) => ({
+      value,
+      color: getColorsForValues(uniqueValues, themeColors)[index],
+    }));
+  }, [uniqueValues, chartsTheme.echartsTheme.color]);
+
+  const option: EChartsCoreOption = {
+    tooltip: {
+      appendToBody: true,
+      formatter: (params: { data: StatusHistoryData; marker: string }) => {
+        return generateTooltipHTML({
+          data: params.data,
+          marker: params.marker,
+          xAxisCategories,
+          yAxisCategories,
+          theme,
+        });
+      },
+    },
+    grid: {
+      top: '5%',
+      bottom: '5%',
+    },
+    xAxis: {
+      type: 'category',
+      data: xAxisCategories,
+      axisLine: {
+        show: false,
+      },
+      splitArea: {
+        show: false,
+      },
+      axisLabel: {
+        hideOverlap: true,
+        formatter: getFormattedStatusHistoryAxisLabel(timeScale?.rangeMs ?? 0, timeZone),
+      },
+    },
+    yAxis: {
+      type: 'category',
+      data: yAxisCategories,
+      axisLine: {
+        show: false,
+      },
+      splitArea: {
+        show: false,
+        interval: 0,
+      },
+      splitLine: {
+        show: false,
+      },
+      axisLabel: {
+        interval: 0,
+      },
+    },
+    visualMap: {
+      show: false,
+      type: 'piecewise',
+      pieces,
+    },
+    series: [
+      {
+        name: 'Status history',
+        type: 'heatmap',
+        coordinateSystem: 'cartesian2d',
+        data: data,
+        label: {
+          show: false,
+        },
+        itemStyle: {
+          borderWidth: 1,
+          borderType: 'solid',
+          borderColor: '#ffffff',
+        },
+        emphasis: {
+          itemStyle: {
+            opacity: 0.5,
+          },
+        },
+      },
+    ],
+  };
+
+  return (
+    <Box sx={{ height: height, overflow: 'auto' }}>
```

Comment:
```suggestion
    <Box style={{height: height }} sx={{ overflow: 'auto' }}>
```
Linked to PR #2408

Echart below can be ignored for now, I need to check if it can be removed safely